base/src/main/java/org/aya/terck/CallResolver.java

```diff
@@ -151,7 +153,7 @@ private Relation compareConArgs(@NotNull ImmutableSeq<Arg<Term>> conArgs, @NotNu
 
   @Override public @NotNull Term pre(@NotNull Term term) {
     // TODO: Rework error reporting to include the original call
-    // term = whnf(term);
+    term = new Expander.ConservativeWHNFer(new TyckState(factory), ImmutableSet.of(caller.ref)).apply(term);
```

Comment:
Should it be `targets` instead of `caller.ref`?

Reply:
Oh good point, in principle we should avoid unfolding any recursive function in the current SCC, and `target` contains all of them, right?

Reply:
I think so. Cc @imkiva